<?xml version="1.0" encoding="UTF-8" ?>
<launch>
    <arg name="robot_type" default="tradr"/>
    <arg name="bag" default="$(dirname)/../data/traversability_local_map_path.bag"/>
    <arg name="rviz" default="true"/>
    <arg name="start" default="0.0"/>
    <arg name="rate" default="1.0"/>

    <param name="use_sim_time" value="true"/>

    <!--
        Workaround for /tf_static problems with rosbag.
        TODO Watch https://github.com/ros/geometry2/issues/181
    -->
    <node name="play_tf_static" pkg="nifti_vision_data" type="play_tf_static"
         args="$(arg bag)" output="log">
        <remap from="~tf_static" to="/tf_static"/>
    </node>


    <node name="rosbag_play_rpz" pkg="rosbag" type="play"
          args="--clock -k -d 3.0 -s $(arg start) -r $(arg rate) $(arg bag)">
    </node>


    <node name="rpz_planner" pkg="rpz_planning" type="rpz_planner" output="screen">
        <rosparam subst_value="true">
            map_frame: map
            robot_frame: base_footprint
            max_age: 1.0
            num_cameras: 6
            num_iters: 25
            map_step: 2
            path_step: 2
            linear_speed: 1.0
            angular_speed: 1.0
<!--            device: cuda:0-->
            device: cpu
        </rosparam>

        <remap from="camera_info_0" to="viz/camera_0/camera_info"/>
        <remap from="camera_info_1" to="viz/camera_1/camera_info"/>
        <remap from="camera_info_2" to="viz/camera_2/camera_info"/>
        <remap from="camera_info_3" to="viz/camera_3/camera_info"/>
        <remap from="camera_info_4" to="viz/camera_4/camera_info"/>
        <remap from="camera_info_5" to="viz/camera_5/camera_info"/>

        <remap from="map" to="traversability"/>
        <remap from="path" to="path"/>
        <remap from="rpz" to="traversability"/>
    </node>


    <node if="$(arg rviz)" name="rviz" pkg="rviz" type="rviz"
          args="-d $(dirname)/tradr.rviz"/>
</launch>
